dataGenerator.js

```diff
@@ -1,74 +1,86 @@
+const makeComments = () => {
+  const comments = [];
+  const commentCount = Math.floor(Math.random() * 10);
 
   for (let x = 0; x < commentCount; x++) {
-    let comment = {};
+    const comment = {};
     comment.username = faker.name.findName();
     comment.avatar = faker.image.cats();
-    // comment.text = loremHipsum({
-    //   count: 2,
-    //   units: 'sentences',
-    //   sentenceLowerBound: 3,
-    //   sentenceUpperBound: 8,
-    //   format: 'plain'
-    // })
-    comment.text = "With hard driving guitars and pounding drums (two kits at the band’s live shows) backing Hull’s pain-laden narratives, Manchester Orchestra has found the “Golden Ticket” into the mainstream.";
-
-    albumData.comments.push(comment);
+    comment.text = loremHipsum({
+      count: 2,
+      units: 'sentences',
+      sentenceLowerBound: 3,
+      sentenceUpperBound: 8,
+      format: 'plain'
+    })
+    comments.push(comment);
   }
+  return comments;
+}
+
+//======= MAIN DATA GENERATOR =======//
+const generateData = async () => {
+  let dataList = [];
+  let fileNameSerial = 0;
+
+  for (let i = 1; i <= entryQty; i++) {
+      let albumData = {};
 
-  dataList.push(albumData);
-
-  if (i !==0 && i % entriesPerFile === 0) { // if i is a multiple of entriesPerFile (10, 20, 30, 40)
-    fileNameSerial++;
-    let notifySerial = fileNameSerial;
-    console.log(`writing file ${notifySerial}...`)
-
-    // fs.writeFile(`./data/testData${fileNameSerial}.json`, JSON.stringify(dataBatch), (err) => {
-    //   if (err) {
-    //     console.log(err);
-    //   } else {
-    //     console.log(`saved file ${notifySerial}`);
-    //   }
-    // })
-    fs.writeFileSync(`./data/testData${fileNameSerial}.json`, JSON.stringify(dataList));
-
-    console.log(`saved file ${notifySerial}`);
-    dataList = [];
-  } else if (i == entryQty) {
-    fileNameSerial++;
-    let notifySerial = fileNameSerial;
-    console.log(`writing file ${notifySerial}...`);
-
-    fs.writeFile(`./data/testData${fileNameSerial}.json`, JSON.stringify(dataList), (err) => {
-      if (err) {
-        console.log(err);
-      } else {
-        console.log(`saved file ${notifySerial}`);
+      albumData.albumID = i;
+      albumData.comments = makeComments();
+
+      dataList.push(albumData);
+
+      if (i !==0 && i % entriesPerBatch === 0 || i == entryQty) { // if i is a multiple of entriesPerBatch
+        fileNameSerial++;
+        let notifySerial = fileNameSerial;
+
+        console.log(`writing batch ${notifySerial}...`)
+
+        if (saveDataTo == 'pg') {
+          const response = await pgdb.insertData(dataList);
+          console.log(`batch ${notifySerial} complete! Rows written: `, response);
+        } else if (saveDataTo == 'fs') {
+          fs.writeFileSync(`./data/testData${fileNameSerial}.json`, JSON.stringify(dataList));
+        } else if (saveDataTo == 'cass') {
+          //TODO: write cassandra db insert
+        }
+
```

Comment:
I also like this method of specifying where you're saving the data! It's concise and readable. 👍